From 78431019b30e3b5dfd31459fd6058d115eaf03d3 Mon Sep 17 00:00:00 2001
From: Oliver Brown <oliver.brown@nxp.com>
Date: Fri, 4 Aug 2017 14:31:35 +0800
Subject: [PATCH 2287/5242] MLK-16137 video: fbdev: add defer probe for mxs
 framebuffer

commit  4bdb612d663f86d598098ba0f5e9fc1a855d4939 from
https://source.codeaurora.org/external/imx/linux-imx.git

When mxs framebuffer probes, the DISPDRV (here is MIPI DSI driver) is likely
not ready, then mxsfb_dispdrv_init() will fail which in turn causes framebuffer
dirver probe fails. Add the defer probe schema for second probe later to fix it.

Signed-off-by: Oliver Brown <oliver.brown@nxp.com>
Signed-off-by: Robby Cai <robby.cai@nxp.com>
Reviewed-by: Fancy Fang <chen.fang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/video/fbdev/mxsfb.c |   11 +++--------
 1 file changed, 3 insertions(+), 8 deletions(-)

diff --git a/drivers/video/fbdev/mxsfb.c b/drivers/video/fbdev/mxsfb.c
index 06fe606..ce431b7 100644
--- a/drivers/video/fbdev/mxsfb.c
+++ b/drivers/video/fbdev/mxsfb.c
@@ -1492,14 +1492,9 @@ static int mxsfb_dispdrv_init(struct platform_device *pdev,
 
 	kfree(setting.dft_mode_str);
 
-	if (IS_ERR(host->dispdrv)) {
-		if (PTR_ERR(host->dispdrv) == -EPROBE_DEFER)
-			return PTR_ERR(host->dispdrv);
-
-		host->dispdrv = NULL;
-		dev_info(dev, "failed to find mxc display driver %s\n",
-			 disp_dev);
-	} else
+	if (IS_ERR(host->dispdrv))
+		return -EPROBE_DEFER;
+	else
 		dev_info(dev, "registered mxc display driver %s\n",
 			 disp_dev);
 
-- 
1.7.9.5

